caps/forms.py

```diff
@@ -56,7 +110,19 @@ def search(self):
             sqs = sqs.narrow(" OR ".join(query_names))
 
         if self.cleaned_data["council_name"]:
-            # narrow makes use of fq rather than q
-            sqs = sqs.narrow("council_name:%s" % self.cleaned_data["council_name"])
+            # limiting to a specific council
+            sqs = sqs.filter(council_name=self.cleaned_data["council_name"])
+
+        # get the highlights so we can do some post-processing on them
+        result = sqs.highlight(**highlighter_kwargs)
+
+        if self.cleaned_data["document_id"]:
+            # adjust the returned object to only include matches for the specific document
+            result = [x for x in result if x.pk == self.cleaned_data["document_id"]]
+
+        if self.cleaned_data["match_method"] != HighlightedSearchForm.MATCH_NORMAL:
+            # improve the highlighting by removing the <mark> tags from the middle of words
+            # for all searches that use exact text
+            result, related_terms = condense_highlights(result, related_terms)
```

Comment:
This is interacting poorly with the pagination which means it does a query for each results page every rather than a single query. I'm not entirely sure how to fix it at the moment.

Reply:
I think I've fixed this by moving the highlighting adjudgment after pagination has happened.